test/e2e/src/test-custom-types.js

```diff
@@ -1,226 +1,237 @@
-    (await t.context.client.u32_fail_on_even({ u32_: 1 })).result,
-    new contract.Ok(1),
-  );
-  t.deepEqual(
-    (await t.context.client.u32_fail_on_even({ u32_: 2 })).result,
-    new contract.Err({ message: "Please provide an odd number" }),
-  );
-});
-
-test("u32", async (t) => {
-  t.is((await t.context.client.u32_({ u32_: 1 })).result, 1); // eslint-disable-line no-underscore-dangle
-});
-
-test("i32", async (t) => {
-  t.is((await t.context.client.i32_({ i32_: 1 })).result, 1); // eslint-disable-line no-underscore-dangle
-});
-
-test("i64", async (t) => {
-  t.is((await t.context.client.i64_({ i64_: 1n })).result, 1n); // eslint-disable-line no-underscore-dangle
-});
-
-test("strukt_hel", async (t) => {
-  const strukt = { a: 0, b: true, c: "world" };
-  t.deepEqual((await t.context.client.strukt_hel({ strukt })).result, [
-    "Hello",
-    "world",
-  ]);
-});
-
-test("strukt", async (t) => {
-  const strukt = { a: 0, b: true, c: "hello" };
-  t.deepEqual((await t.context.client.strukt({ strukt })).result, strukt);
-});
-
-test("simple first", async (t) => {
-  const simple = { tag: "First", values: undefined };
-  const ret = { tag: "First" };
-  t.deepEqual((await t.context.client.simple({ simple })).result, ret);
-});
-
-test("simple second", async (t) => {
-  const simple = { tag: "Second", values: undefined };
-  const ret = { tag: "Second" };
-  t.deepEqual((await t.context.client.simple({ simple })).result, ret);
-});
-
-test("simple third", async (t) => {
-  const simple = { tag: "Third", values: undefined };
-  const ret = { tag: "Third" };
-  t.deepEqual((await t.context.client.simple({ simple })).result, ret);
-});
-
-test("complex with struct", async (t) => {
-  const arg = { tag: "Struct", values: [{ a: 0, b: true, c: "hello" }] };
-  t.deepEqual((await t.context.client.complex({ complex: arg })).result, arg);
-});
-
-test("complex with tuple", async (t) => {
-  const arg = {
-    tag: "Tuple",
-    values: [
-      [
-        { a: 0, b: true, c: "hello" },
-        { tag: "First", values: undefined },
+
+describe("Custom Types Tests", function() {
+  before(async function() {
+    const { client, keypair, contractId } = await clientFor("customTypes");
+    const publicKey = keypair.publicKey();
+    const addr = Address.fromString(publicKey);
+    this.context = { client, publicKey, addr, contractId, keypair };
+  });
+
+  it("hello", async function() {
+    const { result } = await this.context.client.hello({ hello: "tests" });
+    expect(result).to.equal("tests");
+  });
+
+  it("view method with empty keypair", async function() {
+    const { client: client2 } = await clientFor("customTypes", {
+      keypair: undefined,
+      contractId: this.context.contractId,
+    });
+    const { result } = await client2.hello({ hello: "anonymous" });
+    expect(result).to.equal("anonymous");
+  });
+
+  it("woid", async function() {
+    const { result } = await this.context.client.woid();
+    expect(result).to.be.null;
```

Comment:
Here and throughout, I'm curious why you chose to break the one-liners onto two lines. It makes the diff much harder to visually scan. Could do this, right?

```suggestion
    expect(await this.context.client.woid()).to.be.null;
```